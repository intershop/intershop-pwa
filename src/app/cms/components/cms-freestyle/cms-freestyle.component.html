<div *ngIf="pagelet && pagelet.hasParam('HTML')" [ishServerHtml]="pagelet.stringParam('HTML')"></div>
