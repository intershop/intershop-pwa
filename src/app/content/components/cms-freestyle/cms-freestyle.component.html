<div *ngIf="pagelet && pagelet.hasParam('HTML')" [innerHTML]="pagelet.stringParam('HTML') | safeHtml"></div>
