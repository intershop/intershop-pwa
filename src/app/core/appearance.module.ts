import { NgModule } from '@angular/core';
import { NgbPopoverConfig } from '@ng-bootstrap/ng-bootstrap';
import { ToastrModule } from 'ngx-toastr';

import { IconModule } from './icon.module';
import { ThemeService } from './utils/theme/theme.service';

@NgModule({
  imports: [
    IconModule,
    ToastrModule.forRoot({
      closeButton: true,
      timeOut: 3000,
      positionClass: 'toast-top-full-width', // toast-top-center
      preventDuplicates: true,
    }),
  ],
})
export class AppearanceModule {
  constructor(popoverConfig: NgbPopoverConfig, themeService: ThemeService) {
    popoverConfig.placement = 'top';
    popoverConfig.triggers = 'hover';
    popoverConfig.container = 'body';

    themeService.init();
  }
}
