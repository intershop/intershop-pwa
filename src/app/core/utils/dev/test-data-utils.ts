import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';
import { ContentPageTreeHelper } from 'ish-core/models/content-page-tree/content-page-tree.helper';
import { ContentPageTree, ContentPageTreeElement } from 'ish-core/models/content-page-tree/content-page-tree.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}

export function pageTree(elements?: ContentPageTreeElement[]): ContentPageTree {
  const tree = ContentPageTreeHelper.empty();
  if (elements) {
    return elements.reduce((acc, cat) => ContentPageTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
