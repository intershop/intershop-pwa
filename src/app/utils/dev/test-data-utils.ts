import { CategoryTree, CategoryTreeHelper } from 'ish-core/models/category-tree/category-tree.model';
import { Category } from 'ish-core/models/category/category.model';

export function categoryTree(categories?: Category[]): CategoryTree {
  const tree = CategoryTreeHelper.empty();
  if (categories) {
    return categories.reduce((acc, cat) => CategoryTreeHelper.add(acc, cat), tree);
  }
  return tree;
}
