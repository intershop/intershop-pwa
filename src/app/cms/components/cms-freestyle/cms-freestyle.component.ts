import { ChangeDetectionStrategy, Component } from '@angular/core';

import { CMSComponentBase } from '../cms-component-base/cms-component-base';

@Component({
  selector: 'ish-cms-freestyle',
  templateUrl: './cms-freestyle.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSFreestyleComponent extends CMSComponentBase {}
