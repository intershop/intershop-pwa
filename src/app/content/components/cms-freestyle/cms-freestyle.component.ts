import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

import { ContentPageletView } from '../../../models/content-view/content-views';

// tslint:disable-next-line:project-structure
@Component({
  selector: 'ish-cms-freestyle',
  templateUrl: './cms-freestyle.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CMSFreestyleComponent {
  @Input()
  pagelet: ContentPageletView;
}
