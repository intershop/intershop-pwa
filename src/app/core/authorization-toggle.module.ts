import { ModuleWithProviders, NgModule } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { map } from 'rxjs/operators';

import { AuthorizationToggleDirective } from './directives/authorization-toggle.directive';
import { AuthorizationToggleService, checkPermission } from './utils/authorization-toggle/authorization-toggle.service';
import { whenTruthy } from './utils/operators';

@NgModule({
  declarations: [AuthorizationToggleDirective],
  exports: [AuthorizationToggleDirective],
})
export class AuthorizationToggleModule {
  private static permissions = new ReplaySubject<string[]>(1);

  static forTesting(...permissions: string[]): ModuleWithProviders<AuthorizationToggleModule> {
    AuthorizationToggleModule.switchTestingPermissions(...permissions);
    return {
      ngModule: AuthorizationToggleModule,
      providers: [
        {
          provide: AuthorizationToggleService,
          useValue: {
            isAuthorizedTo: (permission: string) =>
              AuthorizationToggleModule.permissions.pipe(
                whenTruthy(),
                map(perms => checkPermission(perms, permission))
              ),
          },
        },
      ],
    };
  }

  static switchTestingPermissions(...permissions: string[]) {
    AuthorizationToggleModule.permissions.next(permissions);
  }
}

export { AuthorizationToggleService } from './utils/authorization-toggle/authorization-toggle.service';
export { AuthorizationToggleGuard } from './guards/authorization-toggle.guard';
