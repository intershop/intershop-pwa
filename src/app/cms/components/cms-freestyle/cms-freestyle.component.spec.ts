import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { SafeHtmlPipe } from 'ish-core/pipes/safe-html.pipe';
import { createSimplePageletView } from 'ish-core/utils/dev/test-data-utils';

import { CMSFreestyleComponent } from './cms-freestyle.component';

describe('Cms Freestyle Component', () => {
  let component: CMSFreestyleComponent;
  let fixture: ComponentFixture<CMSFreestyleComponent>;
  let element: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [CMSFreestyleComponent, SafeHtmlPipe],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(CMSFreestyleComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
  });

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });

  it('should render content if available', () => {
    const pagelet = {
      definitionQualifiedName: 'fq',
      id: 'id',
      displayName: 'name',
      domain: 'domain',
      configurationParameters: { HTML: '<h3>foo</h3>bar' },
    };
    component.pagelet = createSimplePageletView(pagelet);
    expect(() => fixture.detectChanges()).not.toThrow();
    expect(element).toMatchSnapshot();
  });
});
