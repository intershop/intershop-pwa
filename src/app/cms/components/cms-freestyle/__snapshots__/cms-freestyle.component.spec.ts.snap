// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cms Freestyle Component should be created 1`] = `
<div
  id="root0"
>
  
</div>
`;

exports[`Cms Freestyle Component should render content if available 1`] = `
<div
  id="root1"
>
  
  <div>
    <h3>
      foo
    </h3>
    bar
  </div>
</div>
`;
